### PR TITLE
Add the necessary methods to the created models

### PR DESCRIPTION
**Pull request by martinkarugaba**
# Changes made

## Users model
- [ ] Added a method that returns the 3 most recent posts for a given user.

## Posts model
- [ ] A dded a method that updates the posts counter for a user.
- [ ] Added a method which returns the 5 most recent comments for a given post.

## Comments model
Added a method that updates the comments counter for a post.

## Likes model
Added a method that updates the likes counter for a post.

### CODE REVIEW

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @martinkarugaba 🙋‍♀️,  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
 
You have submitted a project that has a README that is not fully documented with all the important information necessary to run the project locally.

Kindly ensure that the `Getting Started` section has all the necessary commands to run the project locally.

- Under the sub-heading `Install` you should have commands like `bundle install` to install gems.
- Under the sub-heading `Usage` you should have commands like `rails db:setup` to create the database and run migrations. And `rails s` to run the project. It is very important to give users a step-by-step guide on how to utilize your project.
- Optionally, you can also include the command `bundle exec rspec` to run all tests under `Run Tests` sub-heading, since you will include specs in your project in the near future.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/post.rb [hunk 1]
```diff
@@ -2,4 +2,12 @@ class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
   has_many :comments
   has_many :likes
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
+
```
Comment (Reem-lab) [inline app/models/post.rb]: - [ OPTIONAL] Amazing job for using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save in comment and like models  to call the #update_comments_counter & update_likes_counter methods, so I highly recommend adding it here as well that every time a new post has been saved to database, then make the `update_post_counter` method private 🎉🔥🚀

**Review by Reem-lab — APPROVED** [review]
# Hi @martinkarugaba,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ]  No Linters error ✅
- [ ]  Kudos for working on the assoications🤩
- [ ]  Great job working on the methods 🚀


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (martinkarugaba) [reply app/models/post.rb]: Thank you @Reem-lab

**Pull request merged**